force-app/main/default/classes/CellOrderTriggerHelper.cls

```diff
@@ -0,0 +1,48 @@
+public with sharing class CellOrderTriggerHelper {
+    public void validateParcelDimensions(List<CellOrder__c> newOrders, Map<ID, SObject> newOrdersMap) {
+        // Collecting all orders
+        List<CellOrder__c> queriedOrders = [
+            SELECT Id,
+                   Cell__r.Width__c, Cell__r.Height__c, Cell__r.Depth__c,
+                   Order__r.Parcel_Width__c, Order__r.Parcel_Height__c, Order__r.Parcel_Depth__c
+            FROM CellOrder__c
+            WHERE Id IN :newOrders
+        ]; 
+        for (CellOrder__c cellOrder : queriedOrders) {
+            
+            // If Cell or Order is empty
+            if (cellOrder.Cell__r.Id == null || cellOrder.Order__r.Id == null) {
```

Comment:
We can simplify it to:

    if (cellOrder.Cell__c == null) {

Note, we can omit checking `cellOrder.Order__c` because it cannot be blank (Master-Detail relationship).